hikari/impl/interaction_server.py

```diff
@@ -81,31 +85,43 @@
 _X_SIGNATURE_TIMESTAMP_HEADER: typing.Final[str] = "X-Signature-Timestamp"
 _CONTENT_TYPE_KEY: typing.Final[str] = "Content-Type"
 _USER_AGENT_KEY: typing.Final[str] = "User-Agent"
+_APPLICATION_OCTET_STREAM: typing.Final[str] = "application/octet-stream"
 _JSON_CONTENT_TYPE: typing.Final[str] = "application/json"
 _JSON_TYPE_WITH_CHARSET: typing.Final[str] = f"{_JSON_CONTENT_TYPE}; charset={_UTF_8_CHARSET}"
 _TEXT_CONTENT_TYPE: typing.Final[str] = "text/plain"
 _TEXT_TYPE_WITH_CHARSET: typing.Final[str] = f"{_TEXT_CONTENT_TYPE}; charset={_UTF_8_CHARSET}"
 
 
 class _Response:
-    __slots__: typing.Sequence[str] = ("_headers", "_payload", "_status_code")
+    __slots__: typing.Sequence[str] = ("_content_type", "_files", "_headers", "_payload", "_status_code")
 
     def __init__(
         self,
         status_code: int,
         payload: typing.Optional[bytes] = None,
         *,
         content_type: typing.Optional[str] = None,
+        files: typing.Sequence[files_.Resource[files_.AsyncReader]] = (),
     ) -> None:
-        self._headers = None
-        if payload or content_type:
-            self._headers = {_CONTENT_TYPE_KEY: content_type or _TEXT_TYPE_WITH_CHARSET}
+        if payload and not content_type:
+            content_type = _TEXT_TYPE_WITH_CHARSET
 
+        self._content_type = content_type
+        self._files = files
+        self._headers = None
         self._payload = payload
         self._status_code = status_code
 
     @property
-    def headers(self) -> typing.Optional[typing.Mapping[str, str]]:
+    def content_type(self) -> typing.Optional[str]:
```

Comment:
Might be worth it getting rid of content_type and enforcing that "payload" is always json